### PR TITLE
Docs: enhance doc for Solaris icons

### PR DESCRIPTION
**Pull request by hannahiss**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_
 
 ### Related issues

Has been splitted from #1929  
~~Should be merged after #2030 in order to have a complete functionality for SVG sprite icons with text color utilities and a good documentation~~
~~Should be merged after #2240 in order to have new warning icon everywhere~~

 ### Description
 
 
In icons documentation (extend > icons):
 - rearrange chapters: Solaris global chapter is not needed since we only talk about Solaris icons
 - remove "Boosted mod: link to Orange Accessibility Guideline" (since it is a Boosted documentation only)
 - use text colors utilities that have been reintroduced by #2030 
 - add warning icon example on white since it is now usable after #2240 
 - update whole documentation
 - remove icomoon mention (not needed anymore with Solaris icons finder)

In Tooltip component (components > tooltips):
 - use text color utility `.text-info` for tooltip icon and remove color from Boosted sprite 

In tables (content > tables):
- change the way to make icons accessible to match icons documentation

Everywhere: 
- change icons names to real ones from Solaris icons finder
- use warning-important-accessible icon since it is now usable (after #2240)

Change icon boosted-cross.svg with the one from Icons finder

Remove icons doc from utilities/colors and point to icons documentation instead

 ### Motivation & Context
 
 
- After Solaris icons finder was released:
  - change documentation to match with Solaris icons finder docs
  - change icons with the ones from finder (viewbox)
  - change boosted-cross.svg with the good icon
- After reintroducing text colors utilities, they can be used for icons, and we have to document it
- Add boosted-warning.svg to allow developers to use bi-colored warning icon 

 ### Types of change
 
 
 
 - New feature (non-breaking change which adds functionality)
 
 
 ### Live previews
 
 
 
 - <https://deploy-preview-2075--boosted.netlify.app/docs/5.3/extend/icons/>
 - <https://deploy-preview-2075--boosted.netlify.app/docs/5.3/components/tooltips#directions>
 - <https://deploy-preview-2075--boosted.netlify.app/docs/5.3/utilities/colors#oranges-colors>
 - <https://deploy-preview-2075--boosted.netlify.app/docs/5.3/content/tables/#with-icons-or-thumbnails>
 
 ### Checklist
 
 
 
 
 #### Contribution
 
 - [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)
 
 #### Accessibility
 
 - [x] My change follows accessibility good practices; I have at least run axe
 
 #### Design
 
 - [x] My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
 - (NA) My change is compatible with responsive display
 
 #### Development
 
 - [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
 - (NA) I have added JavaScript unit tests to cover my changes
 - (NA) I have added SCSS unit tests to cover my changes
 
 #### Documentation
 
 - [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly
 
 ### Checklist (for Core Team only)
 
 - (NA) My change introduces changes to the migration guide
 - (NA) My new component is added in Storybook
 - (NA) My new component is compatible with RTL
 - (NA) Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
 - (NA) Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
 - [ ] Code review
 - [ ] Design review
 - [ ] A11y review
 
 #### After the merge
 
 - [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by hannahiss** [comment]
On hold while text colors utilities aren't merged...

**Comment by julien-deramond** [comment]
Text color utilities are merged so removing the on-hold label

**Comment by julien-deramond** [comment]
FYI, tackled one of the sub-tasks to save you some time via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2075/commits/be5e2e062701b1d0d6a28be5472634a27657b8b2

File: site/content/docs/5.3/extend/icons.md
Comment (louismaximepiton) [inline site/content/docs/5.3/extend/icons.md:152]: Url seems to be broken, here is the good one atm: https://system.design.orange.com/0c1af118d/p/65c68d-solaris-icon-library/b/639b3c

Another approach would consist of describing the way to have access to the good link maybe ? (For instance the good link is at the very bottom of the page)

Maybe the good approach is to do as in the `Learn more about Solaris` link at the top of this page ?

Since this Url will certainly change in the future, I don't really know what's the good approach on this here.

Comment (louismaximepiton) [inline site/content/docs/5.3/extend/icons.md:152]: I don't know if it's feasible with markdown but maybe an ordered list would be better here.

Comment (hannahiss) [reply site/content/docs/5.3/extend/icons.md:152]: URL shouldn't change, even with the new icons finder. I think @julien-deramond checks DSM URLs time to time to be sure they haven't change?

**Review by louismaximepiton — APPROVED** [review]
I'm fine with this, I haven't checked the wording in details but what I read looked fine to me. Let's give the hand to @julien-deramond on this.

**Comment by hannahiss** [comment]
Should be modified and reviewed again with the new Solaris icon finder

**Review by louismaximepiton — COMMENTED** [review]
Some few ideas to discuss. I made some changes in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2075/commits/9551335649196ba09c85fd920b9bab2d018026fb, can you review them ?

File: site/content/docs/5.3/extend/icons.md
Comment (hannahiss) [reply site/content/docs/5.3/extend/icons.md]: discussed with LM: seems too complicated to explain. Advanced users  should be able to do it by themselves

File: site/content/docs/5.3/extend/icons.md
Comment (hannahiss) [reply site/content/docs/5.3/extend/icons.md]: yes you're right

File: site/content/docs/5.3/extend/icons.md
Comment (hannahiss) [reply site/content/docs/5.3/extend/icons.md:123]: I change and use css var to switch color with theme

File: site/static/docs/5.3/assets/img/boosted-warning.svg
Comment (julien-deramond) [inline site/static/docs/5.3/assets/img/boosted-warning.svg:1]: We can probably drop the `fill="currentColor"` part as it won't be used

Comment (julien-deramond) [inline site/static/docs/5.3/assets/img/boosted-warning.svg:1]: We can probably drop the `fill="currentColor"` part as it won't be used:

```suggestion
<svg xmlns='http://www.w3.org/2000/svg' viewBox="0 0 1000 1000">
```

**Review by louismaximepiton — APPROVED** [review]

File: site/assets/scss/_component-examples.scss
Comment (julien-deramond) [inline site/assets/scss/_component-examples.scss:480]: Suggestion to use a more readable value:

```suggestion
  mask-image: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1000 1000'><path d='M553.478 147.416v.017a74.956 74.956 0 0 0-107.08.13v-.021L100 500h150v300a75 75 0 0 0 75 75h350a75 75 0 0 0 75-75V500h150ZM580 537.52V755H362.5a37.5 37.5 0 0 1-37.5-37.5V500h217.5a37.5 37.5 0 0 1 37.5 37.5v.02Z'/></svg>");
```

File: site/assets/scss/_component-examples.scss
Comment (julien-deramond) [inline site/assets/scss/_component-examples.scss:491]: Suggestion to use a more readable value:

```suggestion
  background-image: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1000 1000'><path fill='%23fc0' d='M500.497 125a93.94 93.94 0 0 1 81.09 46.517l328.62 562.5-.008.005a93.709 93.709 0 0 1-81.09 140.983H171.887a93.71 93.71 0 0 1-81.09-140.983l.765-1.326.036-.062 327.8-561.117C436.254 142.707 467.122 125 500.497 125Z'/><path d='M501.507 680.005c-26.233-.002-47.5 21.262-47.502 47.495s21.26 47.5 47.493 47.505a47.5 47.5 0 0 0 47.507-47.5c0-26.233-21.265-47.5-47.498-47.5m-.01-380.007c-26.238 0-47.507 21.27-47.507 47.507 0 .967.037 1.918.094 2.867l15.74 258.716.004.52c.288 17.092 14.355 23.53 31.667 23.53 17.486 0 31.662-6.568 31.67-24.05l15.7-258.121.057-.86a44 44 0 0 0 .082-2.602c0-26.238-21.27-47.507-47.507-47.507' fill='%23000'/></svg>");
```

File: site/static/docs/5.3/assets/img/boosted-sprite.svg
Comment (julien-deramond) [inline site/static/docs/5.3/assets/img/boosted-sprite.svg:12]: I can see a lot of modifications here to have `viewBox="0 0 1000 1000"` versions. Don't we need to update the warning, success, error, etc. icons in `_variables.scss`? (e.g. `$warning-icon-filled` etc.)

File: site/content/docs/5.3/utilities/colors.md
Comment (julien-deramond) [inline site/content/docs/5.3/utilities/colors.md:24]:
```suggestion
To colorize icons, please refer to our [icon color documentation]({{< docsref "extend/icons#modify-icons-color" >}}).
```

File: site/content/docs/5.3/utilities/colors.md
Comment (julien-deramond) [inline site/content/docs/5.3/utilities/colors.md:70]:
```suggestion
Please refer to our [icon color documentation]({{< docsref "extend/icons#modify-icons-color" >}}) to see how to use these text utilities with icons.
```

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:13]:
```suggestion
[Solaris icon library]({{< param icons >}}) is a growing library of icons that are designed by [Orange Design System design team](https://system.design.orange.com/).
```

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:15]: There is probably a missing link here

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:17]:
```suggestion
They are not open-source though and should only be used for Orange projects. Please refer to the [icons license file]({{< param icons_license >}}) for legal information.
```

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:34]:
```suggestion
### SVG sprite

```

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:30]: The link doesn't go to the right page. Plus, we could say "development" instead of "develop"; it sounds weird to me.

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:38]: "text Orange's colors utilities" doesn't go directly to the corresponding section.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Great job @hannahiss!
Some minor suggestions, modifications and questions, and we'll be good to go!

File: site/static/docs/5.3/assets/img/boosted-sprite.svg
Comment (hannahiss) [reply site/static/docs/5.3/assets/img/boosted-sprite.svg:12]: Discussed together: not to be modified for now because the risk is too big to make a mistake and the difference is tiny... 
The differences between Boosted and designs (V6 kit) should be tracked in an issue though (in particular paddings that do not include borders)

File: site/static/docs/5.3/assets/img/boosted-warning.svg
Comment (hannahiss) [reply site/static/docs/5.3/assets/img/boosted-warning.svg]: You're right, it is not consistent... I tried with fill="currentColor", it doesn't work in this context 'external svg image), so I remove it and keep fill="#fc0"

File: site/content/docs/5.3/extend/icons.md
Comment (julien-deramond) [inline site/content/docs/5.3/extend/icons.md:13]:
```suggestion
[Solaris icons library]({{< param icons >}}) is a growing library of icons that are designed by [Orange Design System design team](https://system.design.orange.com/).
```

File: site/static/docs/5.3/assets/img/boosted-sprite.svg
Comment (julien-deramond) [reply site/static/docs/5.3/assets/img/boosted-sprite.svg:12]: Yep, let's create a dedicated issue after having merged this PR to tackle it for the next release

**Review by julien-deramond — APPROVED** [review]

**Comment by julien-deramond** [comment]
Thanks for your comment @hannahiss! It's been taken into account in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2217/commits/d362d1571788856783755efc35280f8c037cf62b.

**Pull request merged**
